tests/Unit/Time/Actions/Test_ChangeStepSize.cpp

```diff
@@ -138,32 +143,31 @@ void check(const bool time_runs_forward,
   using Constant = StepChoosers::Constant<StepChooserUse::LtsStep>;
   MockRuntimeSystem runner{{std::move(time_stepper)}};
 
+  auto choosers =
+      make_vector<std::unique_ptr<StepChooser<StepChooserUse::LtsStep>>>(
+          std::make_unique<Constant>(2. * request),
+          std::make_unique<Constant>(request),
+          std::make_unique<Constant>(2. * request));
+  if (rejector.has_value()) {
+    choosers.emplace_back(std::move(*rejector));
+  }
+
   // Initialize the component
   ActionTesting::emplace_component_and_initialize<component>(
       &runner, 0,
       {TimeStepId(time_runs_forward, 0,
                   time_runs_forward ? time.slab().start() : time.slab().end()),
        TimeStepId(time_runs_forward, 0, time), initial_step_size,
-       initial_step_size,
-       reject_step
-           ? make_vector<std::unique_ptr<StepChooser<StepChooserUse::LtsStep>>>(
-                 std::make_unique<Constant>(2. * request),
-                 std::make_unique<Constant>(request),
-                 std::make_unique<StepRejector>())
-           : make_vector<std::unique_ptr<StepChooser<StepChooserUse::LtsStep>>>(
-                 std::make_unique<Constant>(2. * request),
-                 std::make_unique<Constant>(request),
-                 std::make_unique<Constant>(2. * request)),
-       false, AdaptiveSteppingDiagnostics{1, 2, 3, 4, 5}, std::move(history),
-       1.});
+       initial_step_size, std::move(choosers), false,
+       AdaptiveSteppingDiagnostics{1, 2, 3, 4, 5}, std::move(history), 1.});
 
   ActionTesting::set_phase(make_not_null(&runner), Parallel::Phase::Testing);
   runner.template next_action<component>(0);
   const auto& box = ActionTesting::get_databox<component>(runner, 0);
 
   const size_t index =
       ActionTesting::get_next_action_index<component>(runner, 0);
-  if (reject_step) {
+  if (rejector.has_value()) {
```

Comment:
You just moved the unique ptr out of the rejector above if it had a value. Does it still contain a value, but just a `nullptr`?

Reply:
Yes, that's correct.